pkg/collector/agonsticapi/check.go

```diff
@@ -0,0 +1,283 @@
+  dlerror();
+
+	// Load the function symbol
+	void (*freeResult)(Result*);
+  *(void **)(&freeResult) = dlsym(handle, "FreeResult");
+
+	// Check for errors
+	char *dl_error = dlerror();
+	if (dl_error) {
+		size_t len = strlen("Error loading symbol 'FreeResult': ") + strlen(dl_error) + 1;
+		char *formatted_error = malloc(len);
+		snprintf(formatted_error, len, "Error loading symbol 'FreeResult': %s", dl_error);
+		*error = formatted_error;
+		return;
+	}
+
+
+	// Call the function
+  freeResult(result);
+}
+*/
+import "C"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+	"unsafe"
+
+	"github.com/DataDog/datadog-agent/comp/core/autodiscovery/integration"
+	diagnose "github.com/DataDog/datadog-agent/comp/core/diagnose/def"
+	tagger "github.com/DataDog/datadog-agent/comp/core/tagger/def"
+	"github.com/DataDog/datadog-agent/comp/core/tagger/types"
+	"github.com/DataDog/datadog-agent/pkg/aggregator/sender"
+	"github.com/DataDog/datadog-agent/pkg/collector/agonsticapi/Integrations"
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+	checkid "github.com/DataDog/datadog-agent/pkg/collector/check/id"
+	"github.com/DataDog/datadog-agent/pkg/collector/check/stats"
+	"github.com/DataDog/datadog-agent/pkg/util/hostname"
+)
+
+type agnosticCheck struct {
+	sendermanager sender.SenderManager
+	tagger        tagger.Component
+	name          string
+	libHandle     unsafe.Pointer
+	id            checkid.ID
+}
+
+func newCheck(sendermanager sender.SenderManager, tagger tagger.Component, name string, lib unsafe.Pointer) (check.Check, error) {
+	return &agnosticCheck{
+		sendermanager: sendermanager,
+		tagger:        tagger,
+		name:          name,
+		libHandle:     lib,
+	}, nil
+}
+
+func (c *agnosticCheck) Run() error {
+	var cErr *C.char
+
+	result := C.run_agnostic_check(c.libHandle, &cErr)
+
+	if cErr != nil {
+		errorString := C.GoString(cErr)
+		defer C.free(unsafe.Pointer(cErr))
+
+		errMsg := fmt.Sprintf("failed to execute `run`  function on library %s: err %s", c.name, errorString)
+		return errors.New(errMsg)
+	}
+
+	if result == nil {
+		return fmt.Errorf("library %s execution did not returned any result", c.name)
+	}
+
+	// Extract the FlatBuffer bytes from the Result
+	bufferData := C.GoBytes(unsafe.Pointer(result.data), result.len)
+
+	// Deserialize the FlatBuffer payload
+	payload := Integrations.GetRootAsPayload(bufferData, 0)
+
+	for i := 0; i < payload.MetricsLength(); i++ {
+		var metric Integrations.Metric
+		if payload.Metrics(&metric, i) {
+			name := string(metric.Name())
+			metrciType := string(metric.Type())
+			value := metric.Value()
+			hostn, _ := hostname.Get(context.TODO())
+			tags := []string{}
+
+			for i := range metric.TagsLength() {
+				tags = append(tags, string(metric.Tags(i)))
+			}
+
+			// We need to figure out what tags we need to fetch from the Agent
+			extraTags, err := c.tagger.GlobalTags(types.LowCardinality)
+			if err != nil {
+				fmt.Printf("failed to get global tags: %s\n", err.Error())
```

Comment:
More of a question than a comment: should we drop the metric if we can't global tags?